### PR TITLE
Add WDFN-8 1EP 2x2mm EP0.8x1.2mm

### PR DESCRIPTION
**Pull request by yankee14**
`kicad-symbols`: https://github.com/KiCad/kicad-symbols/pull/2783
`kicad-footprint-generator` for WDFN-8: https://github.com/pointhi/kicad-footprint-generator/pull/612
`kicad-footprints` for WDFN-8: https://github.com/KiCad/kicad-footprints/pull/2457

Datasheet: http://ww1.microchip.com/downloads/en/DeviceDoc/20005474E.pdf#page=25

EP chamfer not working...

WDFN-8-1EP_2x2mm_P0 5mm_EP0 8x1

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -678,6 +678,44 @@ Texas_S-PDSO-N10_EP1.2x2.0mm:
...
+  body_size_x:
+    nominal: 2.0
+    tolerance: 0
+  body_size_y:
+    nominal: 2.0
+    tolerance: 0
+  lead_width_min: 0.2
+  lead_width_max: 0.3
+  lead_len_min: 0.25
+  lead_len_max: 0.35
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:697]: This style is deprecated.
Please use:
```
lead_width: 0.2 ..0.25 ..0.3
lead_len: 0.25 ..0.3 .. 0.35
```

Or

```
lead_width: 0.25 +/- 0.05
lead_len: 0.3 +/- 0.05
```

see https://github.com/pointhi/kicad-footprint-generator/blob/master/scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/Readme.md

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -678,6 +678,44 @@ Texas_S-PDSO-N10_EP1.2x2.0mm:
...
+WDFN-8-1EP_2x2mm_P0.5mm_EP0.8x1.2mm:
+  device_type: 'WDFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'http://ww1.microchip.com/downloads/en/DeviceDoc/20005474E.pdf#page=25'
+  ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  body_size_x:
+    nominal: 2.0
+    tolerance: 0
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:690]: Please write it like
```yaml
body_size_x: 2.0
body_size_y: 2.0
```

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 3]
```diff
@@ -678,6 +678,44 @@ Texas_S-PDSO-N10_EP1.2x2.0mm:
...
+  EP_num_paste_pads: [2, 2]
+
+  pitch: 0.5
+  num_pins_x: 0
+  num_pins_y: 4
+  chamfer_edge_pins: 0.25
+  #pin_count_grid:
+  #pad_length_addition: 0.5
+  #suffix: '_Pad{pad_x:.2f}x{pad_y:.2f}mm_HandSolder'
+  #include_suffix_in_3dpath: 'False'
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:717]: Please remove the unused lines. They just add noise. Same with the commented lines above.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 4]
```diff
@@ -678,6 +678,44 @@ Texas_S-PDSO-N10_EP1.2x2.0mm:
...
+    minimum: 1.1
+    nominal: 1.2
+    maximum: 1.3
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [2, 2]
+
+  pitch: 0.5
+  num_pins_x: 0
+  num_pins_y: 4
+  chamfer_edge_pins: 0.25
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:713]: This is supposed to be used with QFNs. Not for the exposed pad.
Currently the generator does not support a chamfered EP.

**Review by cpresser — CHANGES_REQUESTED** [review]

**Comment by chschlue** [comment]
@cpresser assigned you because you already reviewed it.

**Review by cpresser — APPROVED** [review]

**Pull request merged**